xtask/src/kotlin.rs

```diff
@@ -0,0 +1,166 @@
+                release,
+                profile,
+                src_dir,
+                only_target,
+                package,
+            } => {
+                let profile = profile.as_deref().unwrap_or(if release { "release" } else { "dev" });
+                build_android_library(profile, only_target, src_dir, package)
+            }
+        }
+    }
+}
+
+fn build_android_library(
+    profile: &str,
+    only_target: Option<String>,
+    src_dir: PathBuf,
+    package: Package,
+) -> Result<()> {
+    let root_dir = workspace::root_path()?;
+
+    let package_values = package.values();
+    let package_name = package_values.name;
+    let udl_path = root_dir.join(package_values.udl_path);
+
+    let jni_libs_dir = src_dir.join("jniLibs");
+    let jni_libs_dir_str = jni_libs_dir.to_str().unwrap();
+
+    let kotlin_generated_dir = src_dir.join("kotlin");
+    create_dir_all(kotlin_generated_dir.clone())?;
+
+    let uniffi_lib_path = if let Some(target) = only_target {
+        println!("-- Building for {target} [1/1]");
+        build_for_android_target(target.as_str(), profile, jni_libs_dir_str, package_name)?
+    } else {
+        println!("-- Building for x86_64-linux-android[1/4]");
+        build_for_android_target("x86_64-linux-android", profile, jni_libs_dir_str, package_name)?;
+        println!("-- Building for aarch64-linux-android[2/4]");
+        build_for_android_target("aarch64-linux-android", profile, jni_libs_dir_str, package_name)?;
+        println!("-- Building for armv7-linux-androideabi[3/4]");
+        build_for_android_target(
+            "armv7-linux-androideabi",
+            profile,
+            jni_libs_dir_str,
+            package_name,
+        )?;
+        println!("-- Building for i686-linux-android[4/4]");
+        build_for_android_target("i686-linux-android", profile, jni_libs_dir_str, package_name)?
+    };
+
+    println!("-- Generate uniffi files");
+    generate_uniffi_bindings(&udl_path, &uniffi_lib_path, &kotlin_generated_dir)?;
+
+    println!("-- All done and hunky dory. Enjoy!");
+    Ok(())
+}
+
+fn generate_uniffi_bindings(
+    udl_path: &Path,
+    library_path: &Path,
+    ffi_generated_dir: &Path,
+) -> Result<()> {
+    println!("-- library_path = {}", library_path.to_string_lossy());
+    let udl_file = camino::Utf8Path::from_path(udl_path).unwrap();
+    let out_dir_overwrite = camino::Utf8Path::from_path(ffi_generated_dir).unwrap();
+    let library_file = camino::Utf8Path::from_path(library_path).unwrap();
+
+    uniffi_bindgen::generate_bindings(
+        udl_file,
+        None,
+        vec!["kotlin"],
+        Some(out_dir_overwrite),
+        Some(library_file),
+        false,
+    )?;
+    Ok(())
+}
+
+fn build_for_android_target(
+    target: &str,
+    profile: &str,
+    dest_dir: &str,
+    package_name: &str,
+) -> Result<PathBuf> {
+    cmd!("cargo ndk --target {target} -o {dest_dir} build --profile {profile} -p {package_name}")
```

Comment:
We need to install [`cargo-ndk`](https://github.com/bbqsrc/cargo-ndk) on the CI, as long as all the targets/toolchains passed to this function.

Reply:
I think all the targets are already installed, but yes it's then missing the cargo ndk

Reply:
Then no :-)